template-for-repository/proofs/Makefile.common

```diff
@@ -305,23 +305,26 @@ PROOF_SOURCES ?=
 # report, making the proof run appear to "hang".
 CBMC_TIMEOUT ?= 21600
 
-# Proof writers could add function contracts or loop invariants in
-# CBMC proofs. They should follow three distinct processes:
-# 1. Check whether a function contract is sound, where CHECK_FUNCTION_CONTRACTS
-#    contains the list of functions with contracts to be checked.
-# 2. Replace function calls with their correspondent function contracts,
-#    where USE_FUNCTION_CONTRACTS contains the list of functions with
-#    contracts to be replaced. One must check separately whether a function
-#    contract is sound before replacing it.
-# 3. Apply loop invariants during verification, where APPLY_LOOP_INVARIANTS
-#    contains the list of functions with loop invariants to be applied.
-#    This option is currently unavailable.
+# Proof writers could add function contracts or loop contracts in their source
+# code. These contracts are ignored by default. They may be enabled using three
+# distinct flags:
+# 1. To check whether one or more function contracts are sound with respect to
+#    the function implementation, CHECK_FUNCTION_CONTRACTS should be a list of
+#    function names.
+# 2. To replace calls to certain functions with their correspondent function
+#    contracts, USE_FUNCTION_CONTRACTS should be a list of function names.
+#    One must check separately whether a function contract is sound before
+#    replacing it in calling contexts.
+# 3. To use loop contracts (invariants and variants) for unbounded verification,
+#    USE_LOOP_CONTRACTS should be set to "TRUE".
 CHECK_FUNCTION_CONTRACTS ?= ""
 CBMC_CHECK_FUNCTION_CONTRACTS := $(patsubst %,--enforce-contract %, $(CHECK_FUNCTION_CONTRACTS))
 
 USE_FUNCTION_CONTRACTS ?= ""
 CBMC_USE_FUNCTION_CONTRACTS := $(patsubst %,--replace-call-with-contract %, $(USE_FUNCTION_CONTRACTS))
 
+USE_LOOP_CONTRACTS ?= FALSE
```

Comment:
Why are we setting use loop contracts to false by default and not undefining it by default as we do use and check function contracts?  Turning on functions checking requires just setting the variable, but turning on loop checking requires setting the variable to TRUE.   The implementations should be identical.

Reply:
By undefining do you mean:

```make
USE_LOOP_CONTRACTS ?= 
```

I agree, this could work too.

Reply:
Assuming that's what you meant, I modified the initialization.

Reply:
Changed this to `1` in place of `TRUE`.